### ISSUE TITLE
Delayed Jobs aren't run after cycling Redis

### ISSUE THREAD

**Issue by odigity** [Bogdanp/dramatiq#527]
OS: Ubuntu 20.04
Python: 3.11
Dramatiq: 1.14.0
Redis: "redis:6.2-alpine" Docker image

I created an actor `args` that simply takes an arg and prints it, then ran the following tests:

### Test 1

Tried:  Enqueue job with small delay using `send_with_options(delay=)`.
Result:  Ran successfully after the delay.

### Test 2

Tried:  Enqueue job with small delay, then quickly kill Dramatiq and restart it.
Result:  Still ran job successfully at the appropriate time.

### Test 3

Tried:  Enqueue job with small delay, then quickly kill Redis and restart it.
Result:  Job never runs.  (I can still see the message in Redis key `dramatiq:default.DQ.msgs`.)

When I killed Redis, the worker pool spit out connection errors, but then reconnected.  I tested that by submitting a normal job (not delayed) shortly after, and it ran successfully - proving the workers did re-connect to Redis after Redis came back up.

**Comment by odigity** [comment]
UPDATE:  A few hours later, Dramatiq suddenly decided to run them.  Not sure why.

The DQ and DQ.msgs keys are now empty.

**Comment by Bogdanp** [comment]
When workers pull messages off the queue, they move the messages off their queues onto a separate set (the "acks" set). When a worker crashes (or Redis crashes, same difference), some time (the heartbeat timeout) has to pass until those messages are put back on their respective queues. 

See: https://github.com/Bogdanp/dramatiq/blob/a104a0bf999e6eb655c8d21d9d464e6697c7088b/dramatiq/brokers/redis/dispatch.lua#L110-L153

**Comment by odigity** [comment]
Cool - that was my guess after I saw it recover much later.  Good to know - plus I was testing an extreme case anyway.  (Redis shouldn't go down often.)

I just finished replacing Celery with Dramatiq in our codebase yesterday, and we're testing it in staging right now.

I'm sure I'll have questions in the future - and maybe some contributions.  For now, just glad to have finally escaped that nightmare.
